class StepValueError(ValueError):
    pass

class Iterator:
    def __init__(self, start, stop, step=1 ):
        self.start = start
        self.stop = stop
        self.step = step
        if self.step == 0:
            raise StepValueError('Шаг не может быть равен 0')
        self.pointer = start

    def __iter__(self):
        self.pointer = self.start
        return self

    def __next__(self):
        if self.step > 0:
            if self.stop > 0:
                if self.pointer > self.stop:
                    raise StopIteration()
                point = self.pointer
                self.pointer += self.step
                return point
            if self.stop < 0:
                if self.pointer < self.stop:
                    raise StopIteration()
                point = self.pointer
                self.pointer -= self.step
                return point
        if self.step < 0:
            if self.stop > 0:
                if self.pointer < self.stop:
                    raise StopIteration()
                point = self.pointer
                self.pointer += self.step
                return point
            if self.stop < 0:
                if self.pointer > self.stop:
                    raise StopIteration()
                point = self.pointer
                self.pointer -= self.step
                return point
try:
    iter1 = Iterator(100, 200, 0)
    for i in iter1:
        print(i, end=' ')
except StepValueError:
    print('Шаг указан неверно')

iter3 = Iterator(6, 15, 2)
iter4 = Iterator(5, 1, -1)
iter5 = Iterator(10, 1)

for i in iter3:
    print(i, end=' ')
print()

for i in iter4:
    print(i, end=' ')
print()
for i in iter5:
    print(i, end=' ')
print()
